site/en/docs/android/custom-tabs/howto-custom-tab-from-webview/index.md

```diff
@@ -0,0 +1,32 @@
+---
+layout: "layouts/doc-post.njk"
+title: Open a Custom Tab from a WebView
+date: 2020-02-04
+updated: 2023-05-03
+description: Learn how to open external links in a WebView in a Custom Tab.
+---
+
+When intercepting clicks on links generated by [android:autoLink](https://developer.android.com/reference/android/widget/TextView?cmdf=android%20textview%20auto%20link#attr_android:autoLink) or overriding clicks on links on WebViews, make sure that your application handles the internal links and let's Custom Tabshandle the external ones. 
+
+```java
+WebView webView = (WebView)findViewById(R.id.webview);
+webView.setWebViewClient(new WebViewClient() {
+    @Override
+    public boolean shouldOverrideUrlLoading(WebView view, String url) {
+        return true;
+    }
+
+    @Override
+    public void onLoadResource(WebView view, String url) {
+        if (url.startsWith("http://www.example.com")) {
+            //Handle Internal Link...
+        } else {
+            //Open Link in a Custom Tab
+            Uri uri = Uri.parse(url);
+            new CustomTabsIntent.Builder()
+                  .build()
+                  .launchUrl(context, url));                            
```

Comment:
what is the point of `Uri uri = Uri.parse(url)`? Is it meant to be used instead of `url` on line 28?